### ISSUE TITLE
Settings editor indicators with custom hovers should be link-like

### ISSUE THREAD

**Issue by rzhao271** [microsoft/vscode#153843]
Ref https://github.com/microsoft/vscode/issues/153578#issuecomment-1171243664

Settings editor indicators with custom hovers can be clickable like links. In other words, when a user clicks on an indicator, the custom hover can pop up. 

My plan is to add focus CSS decorations, change the cursor to pointer for those indicators, and also plan on adding tabstops so that the indicators are keyboard-navigable as well.

fyi @isidorn
